giza/agents/integrations/uniswap/nft_manager.py

```diff
@@ -0,0 +1,173 @@
+            position = self.contract.tokenOfOwnerByIndex(user_address, n)
+            positions.append(position)
+        return positions
+
+    def close_position(self, nft_id: int, user_address: str = None):
+        if user_address is None:
+            user_address = self.sender
+        liquidity = self.contract.positions(nft_id)["liquidity"]
+        if liquidity > 0:
+            self.decrease_liquidity(nft_id, liquidity=liquidity)
+            self.collect_fees(nft_id, user_address=user_address)
+            self.contract.burn(nft_id, sender=self.sender)
+
+    def collect_fees(
+        self,
+        nft_id: int,
+        user_address: str = None,
+        amount0_max: int = MAX_UINT_128,
+        amount1_max: int = MAX_UINT_128,
+    ):
+        if user_address is None:
+            user_address = self.sender
+        receipt = self.contract.collect(
+            (nft_id, user_address, amount0_max, amount1_max), sender=self.sender
+        )
+        return receipt
+
+    def decrease_liquidity(
+        self,
+        nft_id: int,
+        liquidity: int = None,
+        amount0Min: int = 0,
+        amount1Min: int = 0,
+        deadline: int = None,
+    ):
+        if liquidity is None:
+            liquidity = self.get_pos_info(nft_id)["liquidity"]
+        if deadline is None:
+            deadline = int(time.time() + 60)
+        receipt = self.contract.decreaseLiquidity(
+            (nft_id, liquidity, amount0Min, amount1Min, deadline), sender=self.sender
+        )
+        return receipt
+
+    def add_liquidity(
+        self,
+        nft_id: int,
+        amount0Desired: int,
+        amount1Desired: int,
+        amount0Min: int = 0,
+        amount1Min: int = 0,
+        deadline: int = None,
+    ):
+        pos = self.contract.positions(nft_id)
+        token0 = Contract(
+            pos["token0"],
+            abi=os.path.join(os.path.dirname(__file__), "assets/erc20.json"),
+        )
+        token1 = Contract(
+            pos["token1"],
+            abi=os.path.join(os.path.dirname(__file__), "assets/erc20.json"),
+        )
+        # give allowances if needed
+        if amount0Desired > token0.allowance(self.sender, self.contract.address):
+            token0.approve(self.contract.address, amount0Desired, sender=self.sender)
+        if amount1Desired > token1.allowance(self.sender, self.contract.address):
+            token1.approve(self.contract.address, amount1Desired, sender=self.sender)
+
+        if deadline is None:
+            deadline = int(time.time() + 60)
+
+        receipt = self.contract.increaseLiquidity(
+            (nft_id, amount0Desired, amount1Desired, amount0Min, amount1Min, deadline),
+            sender=self.sender,
+        )
+        return receipt
+
+    def mint_position(
+        self,
+        pool,
+        lower_price: float,
+        upper_price: float,
+        amount0: int,
+        amount1: int,
+        amount0Min: int = None,
+        amount1Min: int = None,
```

Comment:
Everything that defaults to `None` should have the typehint `int | None`, `str | None` or `float | None`